controller/volume_controller.go

```diff
@@ -954,7 +976,7 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 		// the final state will be determined at the end of the clause
 		if newVolume {
 			v.Status.State = types.VolumeStateCreating
-		} else {
+		} else if v.Status.State != types.VolumeStateDetached {
```

Comment:
@shuo-wu , Here is the fix for the unit test why it changed from detached to detaching. Please help take a look. Thanks.